be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -205,10 +210,19 @@ Status LakePersistentIndex::upsert(size_t n, const Slice* keys, const IndexValue
 
 Status LakePersistentIndex::insert(size_t n, const Slice* keys, const IndexValue* values, int64_t version) {
     RETURN_IF_ERROR(_memtable->insert(n, keys, values, version));
+    KeyIndexSet key_indexes;
+    for (int i = 0; i < n; ++i) {
+        key_indexes.insert(n);
+    }
+    if (_immutable_memtable != nullptr) {
+        RETURN_IF_ERROR(_immutable_memtable->check_not_exist(keys, key_indexes, version));
```

Comment:
what's the meaning of return error after the `_memtable->insert` is done? shouldn't check the existence first and then do the insertion?

Reply:
If duplicated key is found in `_memtable`, `_memtable->insert` will return error. So if `_memtable->insert` returns success, only `immutable_memtable` and `ssts` are needed to be checked.

Reply:
what's the issue this PR try to fix?

let's say key A are not in _memtable, but in immutable_memtable. what is the expected behavior when doing insert(A)? will it report error or success, will the _memtable be polluted even the error reported to caller?

Reply:
In current implementation, `insert` is only used in reload. If error is reported, persistent index is in an error state. The persistent index can not be used any more.

Reply:
done